meta-mender-core/classes/mender-artifactcapsule.bbclass

```diff
@@ -0,0 +1,151 @@
+
+MENDER_ARTIFACT_CAPSULE_DEPENDS ?= ""
+
+MENDER_ARTIFACT_CAPSULE_CLEAR_PROVIDES ?= ""
+
+CAPSULE_GUIDS ?= ""
+
+CAPSULE_TYPE ?= "uefi-firmware"
+
+
+def get_mender_capsule_artifact_provides(d):
+    # This function creates the provides format required by mender artifact to
+    # generate the capsule artifact.
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES needs to be used in the following way.
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES[e2bb9c06-70e9-4b14-97a3-5a7913176111] = "tfa:3"
+    # MENDER_ARTIFACT_CAPSULE_PROVIDES[e2bb9c06-70e9-4b14-97a3-5a7913176222] = "edk2:3"
+    # This gets converted to the  following provides format:
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.name":"tfa",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.version":"3",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.name":"edk2",
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.version":"3",
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_PROVIDES")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    provides = ""
+    for guid in guids:
+        splt = guids[guid].split(":")
+        provides += " --provides " + capsule_type + "." + guid + ".version:" + splt[1]
+        provides += " --provides " + capsule_type + "." + guid + ".name:" + splt[0]
+    return provides
+
+def get_mender_capsule_artifact_depends(d):
+    # This function creates the depends format required by mender artifact to
+    # generate the capsule artifact.
+    # MENDER_ARTIFACT_CAPSULE_DEPENDS needs to be used in the following way.
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176111] = "2"
+    #MENDER_ARTIFACT_CAPSULE_DEPENDS[e2bb9c06-70e9-4b14-97a3-5a7913176222] = "2"
+    # This gets converted to the  following depends format:
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176111.version":"2"
+    # "uefi-firmware.e2bb9c06-70e9-4b14-97a3-5a7913176222.version":"2"
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_DEPENDS")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    depends = ""
+    for guid in guids:
+        depends += " --depends " + capsule_type + "." + guid + ".version:" + guids[guid]
+    return depends
+
+def get_mender_capsule_artifact_clears_provides(d):
+    # This function creates the clears provides format required by mender artifact
+    # to generate the capsule artifact.
+    guids = d.getVarFlags("MENDER_ARTIFACT_CAPSULE_PROVIDES")
+    capsule_type = d.getVar("CAPSULE_TYPE", expand=True)
+    if guids is None:
+        return
+    clears_provides = ""
+    for guid in guids:
+        clears_provides += " --clears-provides " + capsule_type + "." + guid + ".version"
+    return clears_provides
+
+MENDER_PROVIDES = "${@get_mender_capsule_artifact_provides(d)}"
+MENDER_CLEARS_PROVIDES = "${@get_mender_capsule_artifact_clears_provides(d)}"
+MENDER_DEPENDS = "${@get_mender_capsule_artifact_depends(d)}"
```

Comment:
These are quite generic names, and they will be global. I suggest using `CAPSULE` in the name here as well, to avoid ambiguity.

You could also consider a `_MENDER` prefix, which are used in a few other places in meta-mender, since these variables aren't meant to be set directly by the user. That also means they no longer have to be in the `mender-vars.json` file.

Reply:
Exactly, they are not exposed. We often use `_MENDER` as a prefix for such variables to keep them "private", since public variables usually don't start with an underscore.

Reply:
Done